django/db/models/expressions.py

```diff
@@ -312,6 +312,22 @@ def _output_field_or_none(self):
             if not self._output_field_resolved_to_none:
                 raise
 
+    @cached_property
+    def constrains_nulls(self):
```

Comment:
This property means _does this expression requires NULLs to be provided to behave correctly_. Only `IsNull(..., True)` and `Coalesce` have special null handling logic today but this flag allows other expressions to opt in instead of using hardcoded flags.